tensorflow/lite/micro/kernels/xcore/xcore_memory_loader.h

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2021, XMOS Ltd, All rights reserved
+#ifndef XCORE_MEMORY_LOADER_H_
+#define XCORE_MEMORY_LOADER_H_
+
+#include <cstring>
+
+extern "C" {
+#include "nn_operator.h"
+}
+
+namespace tflite {
+namespace micro {
+namespace xcore {
+
+/**
+ * MemoryLoader class
+ *
+ * Reference implementation of the MemoryLoader class.  This class
+ * is used to load data from SwMem (i.e. flash) or ExtMem (i.e. LPDDR).
+ */
+class MemoryLoader {
+ public:
+  MemoryLoader() = default;
+  virtual ~MemoryLoader() = default;
+
+  /**
+   * Load memory from an external memory segment.
+   *
+   * The reference implementation functions much like memcpy. The
+   * primary difference is that this method may utilize the VPU to
+   * accelerate copying memory.
+   *
+   * Override this method to provide an implementation specialized for
+   * your application. For example, memcpy is not necessarily the most
+   * efficient way to load data from flash.  An application developer
+   * may prefer to read directly from flash using a flash library or
+   * RTOS device driver. Additionally, in a multi-threaded application,
+   * one may need to syncronize access to the memory.
+   *
+   * @param[out] dest Pointer to the memory location to copy to
+   * @param[in]  src  Pointer to the memory location to copy from
+   * @param[in]  size Number of bytes to copy
+   */
+  virtual void Load(void *dest, const void *src, size_t size) {
```

Comment:
Should the arguments be `Load(void *&dest, const void *src, size_t size)` to allow implementations to perform the short circuit that `FetchBuffer` does for SRAM addresses? If yes, it should also return `size_t` similarly to `FetchBuffer`.

(Also, the implementation should go into the `.cc`)